runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/SplittableParDoExpander.java

```diff
@@ -91,13 +91,21 @@ public static TransformReplacement createSizedReplacement() {
    * .
    */
   public static TransformReplacement createTruncateReplacement() {
-    return TruncateReplacement.INSTANCE;
+    return SizedReplacement.DRAIN_INSTANCE;
   }
 
   /** See {@link #createSizedReplacement()} for details. */
   private static class SizedReplacement implements TransformReplacement {
 
     private static final SizedReplacement INSTANCE = new SizedReplacement();
+    private static final SizedReplacement DRAIN_INSTANCE = new SizedReplacement().withDrain();
+
+    private boolean isDrain = false;
```

Comment:
It would be nice if we could make this final by using `@AutoBuilder` allowing for:
```suggestion
    private final boolean isDrain;
```